targets/TARGET_STM/us_ticker_16b.c

```diff
@@ -178,20 +169,21 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
 
 void us_ticker_fire_interrupt(void)
 {
-    TimMasterHandle.Instance = TIM_MST;
     HAL_TIM_GenerateEvent(&TimMasterHandle, TIM_EVENTSOURCE_CC1);
 }
 
 void us_ticker_disable_interrupt(void)
 {
-    TimMasterHandle.Instance = TIM_MST;
+    core_util_critical_section_enter();
     __HAL_TIM_DISABLE_IT(&TimMasterHandle, TIM_IT_CC1);
+    core_util_critical_section_exit();
 }
 
 void us_ticker_clear_interrupt(void)
 {
-    TimMasterHandle.Instance = TIM_MST;
-    __HAL_TIM_CLEAR_FLAG(&TimMasterHandle, TIM_FLAG_CC1);
+    core_util_critical_section_enter();
+    __HAL_TIM_CLEAR_IT(&TimMasterHandle, TIM_IT_CC1);
```

Comment:
Just remind that when you gonna change it for 32b to revert it again also for 16b.